### PR TITLE
Add opt-in for sharing analytics data

### PR DESCRIPTION
**Pull request by kevinwallace**
See #60

Screen Shot 2020-05-13 at 6 23 15 PM

### CODE REVIEW

File: Settings/AnalyticsOptIn.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React, {Component} from 'react';
+import {Settings} from 'react-native';
+import TraceTool from './TraceTool';
+
+class AnalyticsOptIn extends Component {
+  constructor(props) {
+    super();
+    this.state = {
+      optIn: !!Settings.get('analyticsOptIn'),
```
Comment (cheng-tan) [inline Settings/AnalyticsOptIn.js:9]: Other settings are saved in asyncStorage, it's better to keep them consistent(there's a helper method in util/asyncstorage). Also Settings only work for iOS.

Comment (kevinwallace) [reply Settings/AnalyticsOptIn.js:9]: I had originally written this using asyncStorage, modeling it off of `Settings/Notification.js`, but there's no apparent way to read those from iOS native code in `AppDelegate.m`, which is where analytics code is initialized.  Do you think it be useful to store in both asyncStorage and `Settings` / `NSUserDefaults`?  I originally wrote it this way, and I wasn't able to see the benefit, so I took it out, but I can put it back if you'd like.  But the use of `Settings` would remain.

Comment (cheng-tan) [reply Settings/AnalyticsOptIn.js:9]: I see. Make sense. Then current way is good.

**Comment by cheng-tan** [comment]
@kevinwallace Hi Kevin, we have multiple piloting coming up and would like some help to get the CovidSafe app completed, wondering if you have bandwidth to pick up some items? Let me know, Thanks.

**Comment by kevinwallace** [comment]
@cheng-tan I appreciate you reaching out to me about this!  Is there a list of tasks you're looking for help on?  I'm uneasy working on location-related functionality, and I see that BLE has been removed (presumably because of the Apple restriction on BLE+location apps?), so I'm not sure there's much left for me to work on, but it's possible that there are still areas where I can help in ways that align with my privacy principles.

**Comment by cheng-tan** [comment]
@kevinwallace We are porting the Android app to React Native and would need help. There are missing features we need to add as well. For example allow user to edit/delete the address in the app, add note/delete contacts in the app, I have mock data for locations and contacts so you don't need to use location service or read from contacts. Can you give me your email so I can add you to our slack channel and provide more details? My email is chengtan2013@gmail.com, if you don't want to leave your email here :)

**Comment by kevinwallace** [comment]
@cheng-tan My concerns are about user privacy generally, not just my own personal privacy, so using mock data doesn't really change things.  I'm uninterested in working on anything that accesses the user's location or contact list.  If BLE functionality returns to iOS, or is reinstated as a standalone app, I would be potentially interested in helping with that.  But I understand that the situation with Apple means that's probably not happening.  If you'd like to email me to discuss further, you can reach me at the email address associated with commit df2b9094aad8952ec3b7f9737ca09e7683c53241.

**Pull request merged**